src/wayland/buffer.rs

```diff
@@ -15,6 +14,49 @@ use wayland_client::{
 
 use super::State;
 
+#[cfg(any(target_os = "linux", target_os = "freebsd"))]
+fn create_memfile() -> File {
+    use nix::sys::memfd::{memfd_create, MemFdCreateFlag};
+
+    let name = unsafe { CStr::from_bytes_with_nul_unchecked("softbuffer\0".as_bytes()) };
+    let fd = memfd_create(name, MemFdCreateFlag::MFD_CLOEXEC)
+        .expect("Failed to create memfd to store buffer.");
+    unsafe { File::from_raw_fd(fd) }
+}
+
+#[cfg(not(any(target_os = "linux", target_os = "freebsd")))]
+fn create_memfile() -> File {
+    use nix::{
+        errno::Errno,
+        fcntl::OFlag,
+        sys::{
+            mman::{shm_open, shm_unlink},
+            stat::Mode,
+        },
+    };
+    use std::iter;
+
+    for _ in 0..=4 {
+        let mut name = String::from("softbuffer-");
+        name.extend(iter::repeat_with(fastrand::alphanumeric).take(7));
+        name.push('\0');
+
+        let name = unsafe { CStr::from_bytes_with_nul_unchecked(name.as_bytes()) };
+        let fd = shm_open(
+            name,
+            OFlag::O_RDWR | OFlag::O_CREAT | OFlag::O_EXCL,
```

Comment:
We created the memfd with the `CLOEXEC` flag on Linux and BSD but not here?

Reply:
I wasn't thinking of this, but apparently that's implied?

>  The FD_CLOEXEC file descriptor flag associated with the new file descriptor is set.

https://pubs.opengroup.org/onlinepubs/9699919799/functions/shm_open.html

Reply:
Hmm okay, I guess we should be explicit about the flag though.

Reply:
I assume passing `O_CLOEXEC` wouldn't cause issues (and would just be ignored), but it's not listed here as one of the flags that can be passed.

Reply:
Okay, I guess you could just mention the implicit flag in a doc comment then.